File: module-api/src/main/java/com/kancho/config/fcm/FcmConfiguration.java
package com.kancho.config.fcm;

import com.google.auth.oauth2.GoogleCredentials;
import com.google.firebase.FirebaseApp;
import com.google.firebase.FirebaseOptions;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.io.ClassPathResource;

import java.io.IOException;

@Configuration
public class FcmConfiguration {
    @Bean
    public FirebaseApp kanchoFirebaseApp() throws IOException {

        FirebaseOptions options = FirebaseOptions.builder()
                .setCredentials(GoogleCredentials.fromStream(new ClassPathResource("kancho-byeolbyeol.json").getInputStream()))
                .build();
        return FirebaseApp.initializeApp(options, "Kancho");
    }
}

File: module-push/src/main/kotlin/com/kancho/config/fcm/FcmConfiguration.kt
package com.kancho.config.fcm

import com.google.auth.oauth2.GoogleCredentials
import com.google.firebase.FirebaseApp
import com.google.firebase.FirebaseOptions
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.io.ClassPathResource

@Configuration
class FcmConfiguration {
    @Bean
    fun kanchoFirebaseApp(): FirebaseApp {

        val options = FirebaseOptions
                .Builder()
                .setCredentials(GoogleCredentials.fromStream(ClassPathResource("kancho-byeolbyeol.json").inputStream))
                .build()

        return FirebaseApp.initializeApp(options, "Kancho")
    }
}